doc/dbus_api.md

```diff
@@ -435,6 +449,127 @@ Arguments:
+Probe()
+Enable(in  ao devices,
+       out u  result)
+Disable(in  ao devices,
+        out u  result)
+SetDiag(in  ao devices,
+        in  b  diag,
+        out u  result)
+~~~
+
+In addition, a `Format()` method is provided, but it's not documented here because it's going to
+change heavily in the short term.
+
+##### Details
+
+###### `Probe` Method
+
+Finds DASDs in the system. Found DASDs are exported with the path
+`/org/opensuse/DInstaller/Storage1/dasds/[0-9]+`.
+
+###### `Enable` Method
+
+~~~
+Enable(in  ao devices,
+       out u  result)
+~~~
+
+Enables the given list of DASDs. See documentation above to understand how the `use_diag` flag of
+the DASDs is affected by this method.
+
+Arguments:
+
+* `in ao devices`: paths of the D-Bus objects representing the DASDs to enable.
+* `out u result`: `0` if all DASDs are successfully enabled. `1` if any of the given paths is invalid (ie. it does not correspond to a known DASD), `2` in case of any other error.
+
+
+###### `Disable` Method
+
+~~~
+Disable(in  ao devices,
+        out u  result)
+~~~
+
+Disables the given list of DASDs.
+
+Arguments:
+
+* `in ao devices`: paths of the D-Bus objects representing the DASDs to disable.
+* `out u result`: `0` if all DASDs are successfully disabled. `1` if any of the given paths is invalid (ie. it does not correspond to a known DASD), `2` in case of any other error.
+
+###### `SetDiag` Method
+
+~~~
+SetDiag(in  ao devices,
+        in  b  diag,
+        out u  result)
+~~~
+
+Sets the `use_diag` attribute for the given DASDs to the given value. See documentation above to
+understand what setting the flag really means (since this follows the same convention than YaST).
+
+Arguments:
+
+* `in ao devices`: paths of the D-Bus objects representing the DASDs to configure.
+* `in b diag`: new value for the flag.
+* `out u result`: `0` if `use_diag` is correctly set for all the requested DASDs. `1` if any of the given paths is invalid (ie. it does not correspond to a known DASD), `2` in case of any other error.
+
+#### `org.opensuse.DInstaller.Storage1.DASD.Device` Interface
+
+This interface is implemented by objects exported at the `/org/opensuse/DInstaller/Storage1/dasds/[0-9]+`
+paths. It provides information about a DASD in the system.
+
+##### Properties
+
+~~~
+Id            readable s
+Enabled       readable b
+DeviceName    readable s
+Formatted     readable b
+Diag          readable b
+Type          readable s
+AccessType    readable s
+PartitionInfo readable s
+~~~
+
+* `Id`: The device channel id (eg. "0.0.0150")
+* `Enabled`: Whether the device is enabled.
+* `DeviceName`: Device name of the DASD in the linux system (eg. "/dev/dasda"). Empty string if the
+  device is not enabled.
+* `Formatted`: whether the device is formatted.
+* `Diag`: Whether the DIAG access method is used (or will be used when the device is enabled).
+* `Type`: The DASD type (eg. EKCD or FBA).
+* `AccessType`: Empty string if unknown. Either "rw" or "ro" otherwise.
```

Comment:
for Type and AccessType I am not sure if unsigned flags won't be better as it can provide better mapping and also easier check for known and unknown methods

Reply:
I though the same, but for the time being I'm only exposing the values coming from YaST. I don't want to operate too much on them.

Reply:
In fact, if the possible values for "AccessType" are really "rw" or "ro" it could be better turned into a "ReadOnly" boolean attribute.

But, as said, for this first version I wanted to stay close to the YaST values since I don't know all the particularities of every attribute and why they are modeled that way in YaST.

Reply:
I added a note about the topic.